### ISSUE TITLE
Fix FN S2138 (`no-undefined-assignment`): We don't flag when assigning value to an object's property

### ISSUE THREAD

**Issue by ilia-kebets-sonarsource** [SonarSource/SonarJS#3715]
Reproducer:

```javascript
const a = { foo: undefined };
```
